### PR TITLE
Resolve custom vendor directories for relative patch files

### PR DESCRIPTION
**Pull request by Caffe1neAdd1ct**
I'm using a repo with patches contained inside pulled in via this patching plugin for composer:

https://github.com/zVPS/laravel-patches/blob/v4.2.22/composer.json

When installing via composer and a custom directory e.g `/vendor-install` via `COMPOSER_VENDOR_DIR=` I get an error as the patches need to be relative to the project root.

This pull request adds the functionality to check for `vendor/` paths and translate to  `COMPOSER_VENDOR_DIR` if specified.

### CODE REVIEW

**Comment by Caffe1neAdd1ct** [comment]
Raised with composer and the view is to fix this here not in composer.

https://github.com/composer/composer/issues/8035

**Comment by stof** [comment]
note that the vendor dir can also be customized through the config, not only from the env variable.

**Comment by Caffe1neAdd1ct** [comment]
Thanks @stof 

Quick example for reference:

```
{
    "config": {
        "vendor-dir": "vendor-install"
    }
}
```

File: src/Netresearch/Composer/Patches/Downloader/Composer.php [hunk 1]
```diff
@@ -43,17 +43,31 @@ protected function getOriginUrl($url) {
...
 	/**
 	 * Download the file and return its contents
 	 * 
 	 * @param string $url The URL from where to download
 	 * @return string Contents of the URL
 	 */
 	public function getContents($url) {
-		return $this->remoteFileSystem->getContents($this->getOriginUrl($url), $url, false);
+	    $path = $this->resolveVendorPath($url);
+		  return $this->cache[$url] = $this->remoteFileSystem->getContents($this->getOriginUrl($url), $path, false);
```
Comment (CybotTM) [inline src/Netresearch/Composer/Patches/Downloader/Composer.php:67]: Where does this `$this->cache[$url]` came from?

**Comment by CybotTM** [comment]
Closing due to being abandoned.

**Pull request closed without merging**